src/Razor/src/Microsoft.AspNetCore.Razor.Language/FileKinds.cs

```diff
@@ -1,12 +1,38 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
+using System.IO;
+
 namespace Microsoft.AspNetCore.Razor.Language
 {
     public static class FileKinds
     {
         public static readonly string Component = "component";
 
         public static readonly string Legacy = "mvc";
+
+        public static bool IsComponent(string fileKind)
+        {
+            // fileKind might be null.
+            return string.Equals(fileKind, FileKinds.Component, StringComparison.OrdinalIgnoreCase);
+        }
+
+        public static string GetFileKindFromFilePath(string filePath)
+        {
+            if (filePath == null)
+            {
+                throw new ArgumentNullException(nameof(filePath));
+            }
+
+            if (string.Equals(".razor", Path.GetExtension(filePath), StringComparison.OrdinalIgnoreCase))
```

Comment:
Guess this is ok to unblock work, but presumably we would want MSBuild \ project system to be the source of truth here? Alternatively we should change the Sdk to complain if it sees files not ending in .razor.

Reply:
We do want that. The problem is that we can't arrive there instantly, and we can't force everyone who calls into Razor to do what we say 😆